bin/oref0_nightscout_check.py

```diff
@@ -92,21 +92,26 @@ def startup_checks(args):
 
 def check_permissions():
     global token_dict
-    pg=token_dict['permissionGroups'][0]
+    pg=[]
+    missing=[]
+
+    for perm_group in token_dict['permissionGroups']:
+        pg.extend(perm_group)
+
     if pg==["*"]: # admin role
         logging.warning("The use of the admin role for token based authentication is not recommended, see https://openaps.readthedocs.io/en/master/docs/walkthrough/phase-1/nightscout-setup.md#switching-from-api_secret-to-token-based-authentication-for-your-rig")
     else:
-        missing=[]
         for perm in ["api:treatments:read", "api:treatments:create", "api:treatments:read", "api:treatments:create", "api:devicestatus:read", "api:devicestatus:create"]:
             logging.debug("Checking %s" % perm)
             if perm not in pg:
-              missing.append(perm)
+                missing.append(perm)
 
-        if len(missing)>0:
-           logging.error("The following permissions are missing in Nightscout: %s" % missing)
-           logging.error("Please follow instructions at https://openaps.readthedocs.io/en/master/docs/walkthrough/phase-1/nightscout-setup.md#switching-from-api_secret-to-token-based-authentication-for-your-rig")
-           sys.exit(1)
-        logging.info("All permissions in Nightscout are ok")
+    if len(missing)>0:
```

Comment:
Why does this need to be promoted out of the else block? If pg==["*"] doesn't that always mean that no perms will be missing by definition?

Reply:
It doesn't :stuck_out_tongue_winking_eye: In my first local iteration, I also promoted the `missing` array, and this went with it. The promotion is no longer needed in the latest iteration.

Reply:
On this note, should I refactor the admin check to `if "*" in pg:` so the excessive permissions are caught even if there are other perms, too?

Reply:
I don't know if anyone relies on that functionality, so I'd rather stick to fixing what's broken unless we get a volunteer to help test any further changes.